model/src/main/kotlin/ScanResults.kt

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2017-2018 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package com.here.ort.model
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import com.fasterxml.jackson.databind.JsonNode
+
+import java.time.Instant
+import java.util.SortedSet
+
+/**
+ * A list of [ScanResult]s for the package identified by [id].
+ */
+data class ScanResults(
+        /**
+         * The [Identifier] of the package these [results] belong to.
+         */
+        val id: Identifier,
+
+        /**
+         * The list of [ScanResult]s.
+         */
+        val results: List<ScanResult>
+)
+
+/**
+ * The result of a single scan of a package.
+ */
+data class ScanResult(
+        /**
+         * Provenance information about the scanned source code.
+         */
+        val provenance: Provenance,
+
+        /**
+         * Details about the used scanner.
+         */
+        val scanner: ScannerSpecification,
+
+        /**
+         * A summary of the scan results.
+         */
+        val summary: ScanSummary,
+
+        /**
+         * The raw output of the scanner.
+         */
+        val rawResult: JsonNode
+)
+
+/**
+ * Provenance information about the scanned source code.
+ */
+data class Provenance(
+        /**
+         * The time when the source code was downloaded.
+         */
+        val downloadTime: Instant,
+
+        /**
+         * The source artifact that was downloaded, or null.
+         */
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        val sourceArtifact: RemoteArtifact? = null,
+
+        /**
+         * The VCS repository that was downloaded, or null.
+         */
+        @JsonInclude(JsonInclude.Include.NON_NULL)
+        val vcsInfo: VcsInfo? = null,
+
+        /**
+         * True if a local directory with no provenance information was scanned. Only serialized if the value is true.
+         */
+        @JsonInclude(JsonInclude.Include.CUSTOM, valueFilter = FalseFilter::class)
+        val localScan: Boolean = false
+)
+
+/**
+ * Details about the used source code scanner.
+ */
+data class ScannerSpecification(
```

Comment:
Maybe call this `ScannerDetails` as it's a bit shorter and you already mention "details" in the docs?

Reply:
Not yet done, but will do when we have decided the naming question above.

Reply:
Done.